x/consensus/keeper/keeper.go

```diff
@@ -69,33 +74,88 @@ func (k Keeper) UpdateParams(ctx context.Context, msg *types.MsgUpdateParams) (*
 	if err != nil {
 		return nil, err
 	}
-
-	paramsProto, err := k.ParamsStore.Get(ctx)
-	if err != nil {
+	if err := cmttypes.ConsensusParamsFromProto(consensusParams).ValidateBasic(); err != nil {
 		return nil, err
 	}
-	params := cmttypes.ConsensusParamsFromProto(paramsProto)
 
-	nextParams := params.Update(&consensusParams)
```

Comment:
this should be maintained, this was a bug fix: https://github.com/cosmos/cosmos-sdk/pull/20347

Reply:
I actually think most of that PR has been lost during merge conflict resolution here